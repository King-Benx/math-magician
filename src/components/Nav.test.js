import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Nav from './Nav';

test('the navigation shows all 3 links', () => {
  render(<BrowserRouter><Nav /></BrowserRouter>);
  const homeLink = screen.getByText('Home');
  const calculatorLink = screen.getByText('Calculator');
  const quoteLink = screen.getByText('Quotes');
  expect(homeLink).toBeTruthy();
  expect(calculatorLink).toBeTruthy();
  expect(quoteLink).toBeTruthy();
});

test('the navigation renders completely', () => {
  const tree = renderer.create(<BrowserRouter><Nav /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
