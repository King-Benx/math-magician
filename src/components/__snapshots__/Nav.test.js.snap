// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the navigation renders completely 1`] = `
<div
  className="nav-bar"
>
  <nav
    className="nav container-width"
  >
    <div>
      <h1>
        Math Magicians
      </h1>
    </div>
    <div
      className="links"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quotes
      </a>
    </div>
  </nav>
</div>
`;
